### ISSUE TITLE
`braces` fixer adds unnecessary empty lines in files where code doesn't start at offset 0

### ISSUE THREAD

**Issue by pilif** [FriendsOfPHP/PHP-CS-Fixer#6647]
## Bug report

given a file where code doesn't start at offset 0 (which is bad, I know), cs-fixer recently started  (through the `braces` fixer) adding random empty lines to the file in front of each indented statement.

If I manually fix the indent, the problem disappears.

### Code snippet that reproduces the problem

```php
<?php

    function foo()
    {
    }
```

running cs-fixer with just the `braces` fixer enabled will produce the following diff

```diff
   1) /path/to/test.php (braces)
      ---------- begin diff ----------
--- /path/to/test.php
+++ /path/to/test.php
@@ -1,5 +1,6 @@
 <?php

+
     function foo()
     {
     }

      ----------- end diff -----------
```

if I remove the 4 space indent at the beginning of all lines, the empty line will not be added.

Also, I can't quite reproduce it yet, but sometimes after making unrelated changes to the file, *another* empty line will be added.

**Comment by pilif** [comment]
incidentally: is there a fixer that removes that unnecessary indentation?

**Comment by julienfalque** [comment]
@pilif This should be fixed by #6667.

> is there a fixer that removes that unnecessary indentation?

`braces` is about to be deprecated in favor of other new rules. This will be documented with #4885. In the meantime you can have a look at the [code of `BracesFixer`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/c1bcf0ce1abd610e8cbd0b309fa3f3085f799535/src/Fixer/Basic/BracesFixer.php#L218-L240) to find the new rules. `statement_indentation` should do what you want.
